### DISCUSSION TITLE
inserendo un ordine, "message": "Il totale dei pagamenti non corrisponde al totale da pagare.",

### DISCUSSION THREAD

**Discussion by Neodixi** [fattureincloud/api#300] [Q&A] [answered]
Buongiorno usando https://developers.fattureincloud.it/api-reference#post-/c/-company_id-/issued_documents provo ad inserire un ordine col seguente Json, ma ottengo sempre l'errore in oggetto. A dire il vero l'errore compare anche con l'esempio proposto.

```json
{
  "data":{
    "type":"order",
    "numeration":"CR809",
    "number":"809",
    "subject":"servizi attività Marzo 23",
    "visible_subject":"servizi attività  Marzo 23",
    "amount_net":850,
    "amount_vat":187,
    "date":"2023-03-14",
    "notes":"<p>come da accordi gi\u00e0 esistenti<\/p>",
    "entity":{
      "id_cliente":"8826222",
      "nome":"cliente azienda."
    },
    "items_list":[
      {
        "product_id":"214",
        "code":"214-Cam",
        "name":"Servizi",
        "qty":"1",
        "description":"",
        "category":"servizi",
        "net_price":850,
        "vat": {
          "id": 0,
          "value": 22,
          "description": ""
        },
        "gross_price":1037,
        "discount":0,
        "discount_highlight":true
      }],
    "payments_list": [
      {
        "amount": 1037,
        "due_date": "2023-03-31",
        "paid_date": null,
        "forfettari_revenue": 1037,
        "payment_terms": {
          "days": 1,
          "type": "standard"
        },
        "status": "not_paid",
      }
    ]
  }
}
```

Grazie

**Comment by emajo** [comment, marked as answer]
Ciao @Neodixi, quell'errore è dovuto al fatto che il totale dei pagamenti non corrisponde al totale del documento, per risolvere questo errore ci sono più opzioni:
- settare manualmente il campo `data.amount_net` e `data.amount_vat` in modo che combacino con il totale dei pagamenti
- settare `data.is_detailed=true`, in questo modo `data.amount_net` e `data.amount_vat` verranno automaticamente settati in base agli items passati (opzione consigliata)

Inoltre abbiamo a disposizione una chiamata apposita per il calcolo dei totali, [qui](https://developers.fattureincloud.it/api-reference#post-/c/-company_id-/received_documents/totals) puoi trovarne la documentazione.

Grazie,
Emanuele.

**Reply by Neodixi** [reply]
Grazie della risposta ma continua a non funzionare 

ecco come ho modifcato il Json 

```json
{
  "data":{
    "type":"order",
    "numeration":"CR809",
    "number":"809",
   "subject":"servizi attività Marzo 23",
   "visible_subject":"servizi attività Marzo 23",
    "is_detailed":true,
    "date":"2023-03-14",
    "notes":"<p>come da accordi gi\u00e0 esistenti<\/p>",
    "entity":{
      "id_cliente":"8826222",
      "nome":"cliente azienda."
    },
    "items_list":[
      {
        "product_id":"214",
        "code":"214-Cam",
        "name":"servizi aziendali",
        "qty":"1",
        "description":"",
        "category":"servizi",
        "net_price":850,
        "vat": {
          "id": 0,
          "value": 22,
          "description": ""
        },
        "gross_price":1037,
        "discount":0,
        "discount_highlight":true
      }],
    "payments_list": [
      {
        "amount": 1037,
        "due_date": "2023-03-31",
        "paid_date": null,
        "forfettari_revenue": 1037,
        "payment_terms": {
          "days": 1,
          "type": "standard"
        },
        "status": "not_paid",
        "payment_account": null
      }
    ]
  }
}
```
Grazie

**Reply by emajo** [reply]
Ciao @Neodixi, con questo body l'errore ritornato è `The entity.name field must not be empty `, questo perché il campo si chiama name e non nome come sta venendo passato.

Ci faccia sapere se risolve,
grazie,
Emanuele.

**Comment by fandex88** [comment]
Ho lo stesso problema
![image](https://user-images.githubusercontent.com/49243706/233090708-321860b3-1711-4c89-8a0e-838e86bd9c01.png)

**Reply by emajo** [reply]
Ciao @fandex88, il problema sembrerebbe essere il campo `payments_list` che sta venendo passato fuori dal campo data.

Grazie,
Emanuele

**Reply by fandex88** [reply]
Si esattamente grazie
